### PR TITLE
[MAR-3537] Add agency edit

### PR DESCRIPTION
**Pull request by malmas18**

### CODE REVIEW

File: src/bundles/ApplicationsAdmin/components/AgencyList/AgencyList.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      loading: true
+    })
+    this.props.loadAgencies()
+      .then(r => {
+        if (r.status === 200) {
+          return r.json()
+        }
+        this.setState({
+          loading: false
+        })
```
Comment (ndavison) [inline src/bundles/ApplicationsAdmin/components/AgencyList/AgencyList.js]: Is this necessary? it looks like the `loading: false` will be set in the next `.then()` anyway.

File: src/bundles/ApplicationsAdmin/components/AgencyList/AgencyList.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  componentDidMount() {
+    this.setState({
+      loading: true
+    })
+    this.props.loadAgencies()
+      .then(r => {
+        if (r.status === 200) {
+          return r.json()
```
Comment (ndavison) [inline src/bundles/ApplicationsAdmin/components/AgencyList/AgencyList.js]: Should this return `Promise.resolve(r.json())`? Or perhaps just set the `agencies` state here and avoid having the extra `.then()`?

File: src/bundles/ApplicationsAdmin/components/AgencyList/AgencyList.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        }
+        this.setState({
+          loading: false
+        })
+        return Promise.reject()
+      })
+      .then(r => {
+        this.setState({
+          loading: false,
+          agencies: r
```
Comment (ndavison) [inline src/bundles/ApplicationsAdmin/components/AgencyList/AgencyList.js]: If the promise is rejected above, then `r` here will not likely be in an expected format. If the `agencies` state change is moved to the previous `.then()` inside the success condition, then this `.then()` can be removed.

File: src/bundles/ApplicationsAdmin/components/Agency/Agency.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+      this.setState({
+        loading: false
+      })
+      return Promise.reject()
+    })
+    .then(r => {
+      this.setState({
+        loading: false,
+        agency: r,
+        saved: new Date()
```
Comment (ndavison) [inline src/bundles/ApplicationsAdmin/components/Agency/Agency.js]: These state changes can probably occur inside the `r.status === 200` condition in the `.then()` above and this `.then()` can be removed, as right now I think these will be called if the Promise is rejected.

File: src/bundles/ApplicationsAdmin/components/Agency/Agency.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+      data.reports = true
+    } else {
+      data.reports = false
+    }
+    if (data.whitelisted === "on") {
+      data.whitelisted = true
+    } else {
+      data.whitelisted = false
+    }
+     data.domains = data.domains.split('\n')
```
Comment (ndavison) [inline src/bundles/ApplicationsAdmin/components/Agency/Agency.js]:
```suggestion
    data.domains = data.domains.split('\n')
```

File: src/bundles/ApplicationsAdmin/components/AgencyList/AgencyList.js (hunk 2 above)
Comment (malmas18) [reply src/bundles/ApplicationsAdmin/components/AgencyList/AgencyList.js]: r.json() returns a promise...

File: src/bundles/ApplicationsAdmin/components/Agency/Agency.js (hunk 4 above)
Comment (malmas18) [reply src/bundles/ApplicationsAdmin/components/Agency/Agency.js]: updated

**Pull request merged**